src/sparseml/transformers/compression/utils/compress_save.py

```diff
@@ -0,0 +1,155 @@
+            save_compressed: bool = False,
+            **kwargs,
+        ):
+            """
+            Wrapper around PreTrainedModel.save_pretrained(), adds functionality for
+            saving models in a compressed format on disk. The compression format is
+            saved to the model's config file
+
+            :param save_directory: output directory to save model to
+            :param sparsity_config: optional sparsity config to compress model with,
+            if no config is provided it will be inferred from the model
+            :param save_compresed: whether or not to compress the model on disk
+            :param kwargs: additional kwargs to pass on to model.save_pretrained
+            """
+            model = model_ref()
+
+            if sparsity_config is not None:
+                # if a sparsity config is provided, always save compressed
+                sparsity_config.fill_config_details(model)
+                save_compressed = True
+            elif save_compressed:
+                # try to infer a sparsity config from the model if none is provided
+                sparsity_config = CompressionConfig.infer_config_from_model(
+                    model, compress=save_compressed
+                )
+
+            if sparsity_config is None:
+                # model is not sparse, save as dense
+                return original_save_pretrained.__get__(model, model_class)(
+                    save_directory, **kwargs
+                )
+
+            # if we've gotten to this point we have a config so we can run compression
+            kwargs["safe_serialization"] = True
+            compressor = ModelCompressor.load_from_registry(
+                sparsity_config.format, config=sparsity_config
+            )
+
+            # state_dict gets passed in as a kwarg for FSDP models
+            state_dict = kwargs.get("state_dict", None)
+            if state_dict is None:
+                state_dict = model.state_dict()
+
+            # make sure we're on the main process when saving
+            if state_dict is not None and len(state_dict) > 0:
+                compressed_state_dict = compressor.compress(state_dict)
+                kwargs["state_dict"] = compressed_state_dict
+
+                original_save_pretrained.__get__(model, model_class)(
+                    save_directory, **kwargs
+                )
+                sparsity_config_data = sparsity_config.dict()
+                config_file_path = os.path.join(save_directory, CONFIG_NAME)
+
+                # add the sparsity config to the model's config file
+                with open(config_file_path, "r") as config_file:
+                    config_data = json.load(config_file)
+                config_data[SPARSITY_CONFIG_NAME] = sparsity_config_data
+                with open(config_file_path, "w") as config_file:
+                    json.dump(config_data, config_file, indent=4, sort_keys=True)
```

Comment:
From checking a few config files, it looks like `indent=2` might be standard? Here's an example https://huggingface.co/neuralmagic/OpenHermes-2.5-Mistral-7B-pruned50/blob/main/config.json